### ISSUE TITLE
High RAM usage with lots of classes with Yolov10

### ISSUE THREAD

**Issue by mazatov** [ultralytics/ultralytics#16601]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm training the model on my own dataset and had no issues training even Yolov10x when I was detecting just one class. However ever since I expanded to many classes (320 classes) I do not have enough RAM to train even Yolov10n. What could be the issue?

I remember I had a similar problem with Yolov7 and there it was the OTA loss that was increasing the RAM usage with the number of classes and by turning that off I could decrease the RAM usage significantly. Is something similar going on here?

**Comment by mazatov** [comment]
Also the RAM seems to go up before the first epoch starts. Once it starts everything is fine and it drops. 

For example I lowered my image size to 320 pixels, workers to 0, batch to 1 and it finally went through. Once the first epoch kicked of on GPU the RAM dropped to super low percentage. It seems like it's something to do with initialization.

**Comment by Y-T-G** [comment]
What's the training code or command you're using?

**Comment by mazatov** [comment]
I'm using command line

yolo task=detect mode=train epochs=500 batch=1 plots=True
model=yolov10s.yaml data=data.yaml imgsz= 640 device=0 name="$TRAIN_NAME"
scale=0.5 translate=0.2 mixup=0.15 mosaic=1.0 fliplr=0
fi

**Comment by Y-T-G** [comment]
Does the RAM go up or VRAM?

**Comment by mazatov** [comment]
RAM. My gpu at that time is at 13% load.

**Comment by mazatov** [comment]
![Screenshot 2024-10-02 105337](https://github.com/user-attachments/assets/6328dcb5-fd09-4534-9746-c84f4e7dc0cd)

Here's example of me running on extremely low parameters. The only thing that is high is number of classes. 
```
WEIGHTS="yolov10n.yaml"
BATCH=1
WORKERS=0
IMG_SIZE=320
```

**Comment by Y-T-G** [comment]
Does it happen with other models?

**Comment by mazatov** [comment]
I've only tested it with Yolov10 for now.

**Comment by Y-T-G** [comment]
You can check YOLOv8n or YOLO11n.
